### DISCUSSION TITLE
How to access Control API with JWT

### DISCUSSION THREAD

**Discussion by einaraglen** [bluenviron/mediamtx#3271] [Questions] [answered]
### Question

I've tried adding header "Authorization": "Bearer TOKEN-HERE"

adding query params my-host:9997/v3/paths/list?jwt=TOKEN-HERE

and nothing seems to work, i just get 401, now if i disable JWT auth all together, i can access the API.

I cannot for the life of me find where you spesify how to access the control API using JWT.

for refrence when i try to use the auth token as header, this is what is printed by the MediaMTX instance:

`
2024/04/18 12:42:16 DEB [API] [conn MY-IPADDRESS] [c->s] GET /v3/paths/list HTTP/1.1
Host: MY-HOST
Accept: */*
Accept-Encoding: gzip, deflate, br
Authorization: Bearer TOKEN-HERE
Cache-Control: no-cache
Connection: keep-alive
Postman-Token: 1dd1a7fa-9492-48b7-825a-0319cd165cb1
User-Agent: PostmanRuntime/7.36.1

2024/04/18 12:42:16 DEB [API] [conn MY-IPADDRESS] [s->c] HTTP/1.1 401 Unauthorized
Server: mediamtx
Www-Authenticate: Basic realm="mediamtx"
`

I am using keycloak to generate these tokens and they seem to be containing the correct permissions:

`
"mediamtx_permissions": [
    {
      "action": "publish",
      "paths": "all"
    },
    {
      "action": "read",
      "paths": "all"
    },
    {
      "action": "api",
      "paths": "all"
    }
  ],
`

also in `mediamtx.yml` i have added `authMethod: internal` and `api: yes`

**Comment by aler9** [comment, marked as answer]
Hello, this is a bug that was fixed with #3253

**Reply by einaraglen** [reply]
Hi, when do you think a new docker image for this fix will be created?

should i rather make it from source myself?

**Comment by rfscheidt** [comment]
Hello,

I encountered the same error and submitted a pull request to @aler9, which was accepted. However, I'm still working on fixing it for HLS. Can anyone help?

**Comment by rfscheidt** [comment]
@aler9 The first time the package passes through here, it retrieves the ?jwt=XXX, but when it passes again with mp4, stream, etc., it doesn't return the jwt in the URL.

<img width="872" alt="image" src="https://github.com/bluenviron/mediamtx/assets/7602639/888f2ba5-d71e-4458-8540-7584a24c71f9">

Can you give me any suggestions on where to put the JWT again to return in the next request?

**Reply by aler9** [reply]
@rfscheidt the question is not trivial - HLS playlists have to be hacked in order to inject incoming query parameters into subplaylists and media segments. I've created a dedicated issue here: #3275

[a comment by an automated account is not shown]
